logstash-core/src/main/java/org/logstash/plugins/discovery/PluginRegistry.java

```diff
@@ -95,24 +103,51 @@ private void discoverPlugins() {
                 }
             }
         }
+
+        // after loaded all plugins, check if aliases has to be provided
+        addAliasedPlugins(PluginType.FILTER, filters);
+        addAliasedPlugins(PluginType.OUTPUT, outputs);
+        addAliasedPlugins(PluginType.INPUT, inputs);
+        addAliasedPlugins(PluginType.CODEC, codecs);
     }
 
-    public Class<?> getPluginClass(PluginLookup.PluginType pluginType, String pluginName) {
-        if (pluginType == PluginLookup.PluginType.FILTER) {
-            return getFilterClass(pluginName);
+    private <T> void addAliasedPlugins(PluginType type, Map<String, Class<T>> pluginCache) {
+        final Map<String, Class<T>> aliasesToAdd = new HashMap<>();
+        for (Map.Entry<String, Class<T>> e : pluginCache.entrySet()) {
+            final String realPluginName = e.getKey();
+                final Optional<String> alias = aliasRegistry.aliasFromOriginal(type, realPluginName);
+                if (alias.isPresent()) {
+                    final String aliasName = alias.get();
+                    if (!pluginCache.containsKey(aliasName)) {
+                        // no real plugin with same alias name was found
+                        aliasesToAdd.put(aliasName, e.getValue());
+                    }
+                }
         }
-        if (pluginType == PluginLookup.PluginType.OUTPUT) {
-            return getOutputClass(pluginName);
+        for (Map.Entry<String, Class<T>> e : aliasesToAdd.entrySet()) {
+            pluginCache.put(e.getKey(), e.getValue());
         }
-        if (pluginType == PluginLookup.PluginType.INPUT) {
-            return getInputClass(pluginName);
+    }
+
+    public Class<?> getPluginClass(PluginType pluginType, String pluginName) {
+        if (aliasRegistry.isAlias(pluginType, pluginName)) {
+            final String typeStr = pluginType.name().toLowerCase();
+            LOGGER.info("Plugin {} is aliased as {}", typeStr + "-" + pluginName,
+                    typeStr + "-" + aliasRegistry.originalFromAlias(pluginType, pluginName));
         }
```

Comment:
This logger message is a bit confusing when a user requests a plugin that is both a concrete installed plugin and an alias to another plugin.

Let's remove it here and instead add something similar when the alias is being added to the registry.

```suggestion
```